components/adblock_rust_ffi/src/lib.rs

```diff
@@ -152,12 +152,17 @@ pub unsafe extern "C" fn engine_match(
     *did_match_exception |= blocker_result.exception.is_some();
     *did_match_important |= blocker_result.important;
     *redirect = match blocker_result.redirect {
-        Some(Redirection::Resource(x)) => match CString::new(x) {
+        Some(x) => match CString::new(x) {
+            Ok(y) => y.into_raw(),
+            _ => ptr::null_mut(),
+        },
+        None => ptr::null_mut(),
```

Comment:
would `.unwrap_or(ptr::null_mut())` be more clear here than the nested match? maybe not.